velox/expression/tests/CastExprTest.cpp

```diff
@@ -2041,5 +2059,41 @@ TEST_F(CastExprTest, identicalTypes) {
   ASSERT_EQ(result.get(), data->childAt(0).get());
 }
 
+TEST_F(CastExprTest, skipCastEvaluation) {
+  // Inputs to error_on_odd are even, odd.
+  // Input to cast is an UNKNOWN vector which is not supported.
+  {
+    auto data = makeRowVector({
+        makeFlatVector<int>(10, [&](auto row) { return row; }),
+    });
+
+    VELOX_ASSERT_THROW(
+        evaluate("try(cast(error_on_odd(c0) as INTEGER))", data),
+        "not a scalar type! kind: UNKNOWN");
+  }
+
+  // All inputs to error_on_odd are odd.
+  // All inputs to cast are errors, we skip evaluation.
+  {
+    auto data = makeRowVector({
+        makeFlatVector<int>(10, [&](auto row) { return row * 2 + 1; }),
+    });
+
+    auto result = evaluate("try(cast(error_on_odd(c0) as INTEGER))", data);
+    ASSERT_EQ(BaseVector::countNulls(result->nulls(), result->size()), 10);
+  }
+
+  // The result vector can shrink since we remove the error rows.
```

Comment:
nit: maybe add
`Ensure trailing rows that are marked as error are handled correctly as they can result in intermediate result vectors of smaller size`